File: Tests/src/pages/LoginPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import utilities.Constants;

public class LoginPage extends BasePage {
	@FindBy(id = "login")
	public WebElement loginFld;
	
	@FindBy(id = "password")
	public WebElement passwordFld;
	
	@FindBy(id = "loginBtn")
	public WebElement loginBtn;
	
	@FindBy(id = "errorMessage")
	public WebElement errorMessage;

	@FindBy(xpath = "//div[text() = 'Register']")
	private WebElement register;
	
	public LoginPage(WebDriver driver) {
		super(driver);
		PageFactory.initElements(this.driver, this);
	}

	public RegisterPage openRegistration() {
		register.click();
		
		return new RegisterPage(driver);
	}
	
	public LoginPage deleteAllUsers() {
		driver.get(Constants.BaseUrl + "/Register/DeleteAllUsers");
		driver.get(Constants.BaseUrl);

		return this;
	}
	
	public DepositPage login() {
		return login(Constants.Login, Constants.Password);
	}
	
	public DepositPage login(String login, String password) {
		loginFld.sendKeys(login);
		passwordFld.sendKeys(password);
		loginBtn.click();

		return new DepositPage(driver);
	}
}

File: Tests/pages/LoginPage.py
from time import sleep
from selenium.webdriver.common.by import By

from Constants import Constants
from pages.BasePage import BasePage
from pages.DepositPage import DepositPage


class LoginPage(BasePage):
    def __init__(self, driver):
        super().__init__(driver)
        self.driver.get(Constants.URL + "/Login")
        sleep(1)

    def open_registration(self):
        from pages.RegisterPage import RegisterPage
        self.find_element(By.XPATH, "//div[text() = 'Register']").click()
        return RegisterPage(self.driver)

    def remind_password(self):
        from pages.RemindPasswordFrame import RemindPasswordFrame
        btn = self.find_element(By.ID, "remindBtn")
        return RemindPasswordFrame(self.driver, btn)

    def login(self, login=Constants.LOGIN, password=Constants.PASSWORD):
        self.set_input_value(login, "login")
        self.set_input_value(password, "password")
        self.find_element(By.ID, "loginBtn").click()
        sleep(1)

        if self.element_exists(By.ID, "errorMessage"):
            return self.find_element(By.ID, "errorMessage").text

        return DepositPage(self.driver)
